editor_drafts/draft-prorock-spice-cose-sd-cwt.md

```diff
@@ -124,93 +138,104 @@ The parameters necessary to support these processes can be obtained using transp
+The salt acts as a blinding factor, preventing a Verifier of an SD-CWT from learning claims that were not intentionally disclosed by a Holder.
+A confirmation claim `cnf (8)` MUST be present in the CWT Claimset.
+The `sd_kbt` MUST NOT be set by the Issuer, and MUST be set by the Holder, and is therefore marked optional in the following normative defintion of SD-CWT in CDDL:
 
 ``` cddl
+
+digested-salted-disclosed-claim = bstr; 
+salted-disclosed-claim = salted-claim / salted-element
+salted-claim = [
+  bstr .size 16,  ; 128-bit salt
+  (int / text),   ; claim name
+  any             ; claim value
+]
+salted-element = [
+  bstr .size 16, ; 128-bit salt
+  any            ; claim value
+]
+sd-cwt-cnf = COSE_Key / Encrypted_COSE_Key / kid
 sd-cwt = [
   protected,
   unprotected: {
-    ?(sd_claims: TBD): bstr .cbor [ + claim-pair],
-    ?(sd_kbt: TBD): bstr .cbor sd-cwt-kbt,
+    ?(sd_claims: TBD1): bstr .cbor [ + salted-disclosed-claim ],
+    ?(sd_kbt: TBD2): bstr .cbor sd-cwt-kbt,
   },
   payload :  bstr .cbor {
-    &(iss: 1) => tstr, ; "https://issuer.example"
-    &(sub: 2) => tstr, ; "https://device.example"
-    &(aud: 3) => tstr, ; "https://verifier.example"
-    &(exp: 4) => int,  ; 1883000000
-    &(nbf: 5) => int,  ; 1683000000
-    &(iat: 6) => int,  ; 1683000000
+    ?(iss: 1) => tstr, ; "https://issuer.example"
+    ?(sub: 2) => tstr, ; "https://device.example"
+    ?(aud: 3) => tstr, ; "https://verifier.example"
+    ?(exp: 4) => int,  ; 1883000000
+    ?(nbf: 5) => int,  ; 1683000000
+    ?(iat: 6) => int,  ; 1683000000
     &(cnf: 8) => sd-cwt-cnf,  ; 1683000000
 
-    ; sd-cwt new claims
-    &(sd_alg: TBD) => int,            ; -16 for sha-256
-    &(redacted_keys: TBD) => [ bstr ] ; redacted map key
-    &(swversion: 271) => [              ; example array
-      ...,
-      {
-        &(redacted_element: TBD) => bstr ; redacted array element
-      }
-      ...
-    ]      
+    ?(sd_alg: TBD4) => int,             ; -16 for sha-256
+    ?(redacted_keys: TBD5) => [         ; redacted map keys
+      digested-salted-disclosed-claim 
+    ],
+
+    ; redaction in an example map value that is an array 
+    &(example-array-key: -65537) => [  
+      123,
+      { ; redacted array element
+        &(redacted_element: TBD6) =>
+        digested-salted-disclosed-claim 
+      },
+      789,
+      { ; redacted array element
+        &(redacted_element: TBD6) =>
+        digested-salted-disclosed-claim 
+      },
+    ]
   }
   signature : bstr,
 ]
 ```
 
-Disclosures are structured as a "sd-cwt-claim-pair" with a 32 bit salt, and the byte string of the disclosed value.
-
-``` cddl
-sd-cwt-claim-pair = [
-  ; The RECOMMENDED minimum length 
-  ; of the randomly-generated portion of the salt is 128 bits
-  uint .size 16,  
-  bstr           ; disclosed value
-]
-```
+As described above, an SD-CWT is a CWT with claims require confirmation and support selective disclosure.
```

Comment:
I'm failing to parse this sentence. Seems like a word may be missing. Maybe:
```suggestion
As described above, an SD-CWT is a CWT with claims that require confirmation and support selective disclosure.
```
But I'm not sure if it is the `claims` that require confirmation and support selective disclosure (as written above), 

or whether it should be the `SD-JWT` that require**s** claims and support**s** selective disclosure.